src/App.tsx

```diff
@@ -139,8 +139,9 @@ const App = () => {
             {values.map((value, index) => {
               return (
                 <Validation
+                  // biome-ignore lint/suspicious/noArrayIndexKey: items order does not change
                   key={`val${index}`}
-                  schema={evaluatedSchema}
+                  schema={evaluatedSchema as zod.ZodSchema}
```

Comment:
I think you should assert the type inside the validateSchema function to data variable